dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java

```diff
@@ -361,6 +364,12 @@ public T deleteHostedVersion(@ApiParam(hidden = true) @Auth User user,
         checkEntry(entry);
         checkUserCanUpdate(user, entry);
         checkHosted(entry);
+        // If the version that's about to be deleted is the default version, unset it
+        if (entry.getActualDefaultVersion().getName().equals(version)) {
+            Optional<U> max = entry.getWorkflowVersions().stream().filter(v -> !Objects.equals(v.getName(), version))
+                    .max(Comparator.comparingLong(ver -> ver.getDate().getTime()));
+            entry.setActualDefaultVersion(max.orElse(null));
```

Comment:
I feel like we should give the user with a warning that they're deleting the default version so we're setting the new default to latest. Maybe we should log this message to surface in the UI?

Reply:
@garyluu please create a follow-up ticket

Reply:
https://github.com/dockstore/dockstore/issues/3381